daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -264,6 +265,111 @@ packagingTests = testGroup "packaging"
                 _ -> assertFailure "Expected JSON object in response body"
               -- waitForProcess' will block on Windows so we explicitly kill the process.
               terminateProcess startPh
+     , testCase "hot-reload" $ withTempDir $ \tmpDir -> do
+        let projDir = tmpDir </> "hotreload"
+        createDirectoryIfMissing True (projDir </> "daml")
+        writeFileUTF8 (projDir </> "daml.yaml") $ unlines
+          [ "sdk-version: " <> sdkVersion
+          , "name: hotreload"
+          , "version: \"1.0\""
+          , "source: daml"
+          , "dependencies:"
+          , "  - daml-prim"
+          , "  - daml-stdlib"
+          , "  - daml-script"
+          , "parties:"
+          , "- Alice"
+          , "init-script: Main:init"
+          , "sandbox-options:"
+          , "  - --wall-clock-time"
+          ]
+        writeFileUTF8 (projDir </> "daml/Main.daml") $ unlines
+          [ "module Main where"
+          , "import Daml.Script"
+          , "template T with p : Party where signatory p"
+          , "init : Script ()"
+          , "init = do"
+          , "  alice <- allocatePartyWithHint \"Alice\" (PartyIdHint \"Alice\")"
+          , "  alice `submit` createCmd (T alice)"
+          , "  pure ()"
+          ]
+        sandboxPort :: Int <- fromIntegral <$> getFreePort
+        jsonApiPort :: Int <- fromIntegral <$> getFreePort
+        let startProc = (shell $ unwords
+              [ "daml"
+              , "start"
+              , "--start-navigator"
+              , "no"
+              , "--sandbox-port"
+              , show sandboxPort
+              , "--json-api-port"
+              , show jsonApiPort
+              ]) {std_in = CreatePipe, cwd = Just projDir}
+        withCurrentDirectory projDir $
+          withCreateProcess startProc $ \startStdIn _ _ startPh ->
+            race_ (waitForProcess' startProc startPh) $
+             do
+              -- The hard-coded secret for testing is "secret".
+              let token =
+                    JWT.encodeSigned
```

Comment:
Don't we have a helper function for creating such tokens in one of the tests you wrote recently?

Reply:
The new http json api tests create the sandbox and http json api with a configurable secret. But there is no function exported for creating a token. I think the Assistant/IntegrationTests.hs are quite messy right now and would benefit from factoring out a few commonly used functions such as the token creation.

Reply:
Ok, thanks for the clarification. Do you have some concrete ideas for such refactorings? If so, please go ahead. If not, let's chat about it.